.travis.yml

```diff
@@ -1,47 +1,64 @@
 language: android
-jdk: oraclejdk8
-
-# Choosing faster environment since it's a full VM.
-# https://docs.travis-ci.com/user/reference/overview/
-sudo: required
-dist: trusty
+jdk: oraclejdk7
 
 env:
   matrix:
-    # Notice matrix environment variables are not available during the install phase for android >
-    # components.
-    # These CANNOT BE USED in the android > components section, only in before_script and script.
-    - TEST_TYPE=unit
-    - TEST_TYPE=instrumentation EMULATOR_TAG=android EMULATOR_API=16 ABI=armeabi-v7a
-    - TEST_TYPE=instrumentation EMULATOR_TAG=android EMULATOR_API=19 ABI=armeabi-v7a
-    - TEST_TYPE=instrumentation EMULATOR_TAG=android EMULATOR_API=21 ABI=armeabi-v7a
+  - EMULATOR_API=16
+  - EMULATOR_API=19
+  - EMULATOR_API=21
```

Comment:
Could we add some newer devices to this mix? We've been holding off since it has been failing, but in an ideal world we'd have tests for maybe 23 and something newer.

Reply:
Great suggestion, I added API level 24 to the matrix so now CI tests four devices:
* API Level 16 Jellybean
* API Level 19 Kitkat
* API Level 21 Lollipop
* API Level 24 Nougat